fenix/app/src/main/java/org/mozilla/fenix/compose/tabstray/TabGridItem.kt

```diff
@@ -94,95 +99,115 @@ fun TabGridItem(
+                elevation = 0.dp,
+                shape = RoundedCornerShape(dimensionResource(id = R.dimen.tab_tray_grid_item_border_radius)),
+                border = BorderStroke(1.dp, FirefoxTheme.colors.borderPrimary),
             ) {
-                Row(
-                    modifier = Modifier
-                        .fillMaxWidth()
-                        .wrapContentHeight(),
+                Column(
+                    modifier = Modifier.background(FirefoxTheme.colors.layer2),
                 ) {
-                    Favicon(
-                        url = tab.content.url,
-                        size = 16.dp,
-                        modifier = Modifier
-                            .align(Alignment.CenterVertically)
-                            .padding(start = 8.dp),
-                    )
-
-                    HorizontalFadingEdgeBox(
+                    Row(
                         modifier = Modifier
-                            .weight(1f)
-                            .wrapContentHeight()
-                            .requiredHeight(30.dp)
-                            .padding(7.dp, 5.dp)
-                            .clipToBounds(),
-                        backgroundColor = FirefoxTheme.colors.layer2,
-                        isContentRtl = BidiFormatter.getInstance().isRtl(tab.content.title),
+                            .fillMaxWidth()
+                            .wrapContentHeight(),
                     ) {
-                        Text(
-                            text = tab.toDisplayTitle().take(MAX_URI_LENGTH),
-                            fontSize = 14.sp,
-                            maxLines = 1,
-                            softWrap = false,
-                            style = TextStyle(
-                                color = FirefoxTheme.colors.textPrimary,
-                                textDirection = TextDirection.Content,
-                            ),
-                        )
-                    }
-
-                    if (!multiSelectionEnabled) {
-                        Icon(
-                            painter = painterResource(id = R.drawable.mozac_ic_close),
-                            contentDescription = stringResource(id = R.string.close_tab),
-                            tint = FirefoxTheme.colors.iconPrimary,
+                        Favicon(
+                            url = tab.content.url,
+                            size = 16.dp,
                             modifier = Modifier
-                                .clickable { onCloseClick(tab) }
-                                .size(24.dp)
                                 .align(Alignment.CenterVertically)
-                                .testTag(TabsTrayTestTag.tabItemClose),
+                                .padding(start = 8.dp),
                         )
+
+                        HorizontalFadingEdgeBox(
+                            modifier = Modifier
+                                .fillMaxWidth()
```

Comment:
was this intentionally changed from `.weight(1f)` to `.fillMaxWidth()`? The close icon isn't showing anymore (the preview captures this visual bug), which is likely also causing the UI test failure.